### PR TITLE
Simplify pagination

### PR DESCRIPTION
**Pull request by staheri14**
Closes https://github.com/status-im/nim-waku/issues/484

### CODE REVIEW

File: waku/v2/protocol/waku_store/waku_store.nim [hunk 1]
```diff
@@ -289,54 +289,80 @@ proc paginate*(list: seq[IndexedWakuMessage], pinfo: PagingInfo): (seq[IndexedWa
...
+      var endIndex = cursorIndex - 1
+      # for the initial query, include the message pointing by the cursor
+      if isInitialQuery:  
+        endIndex = cursorIndex
+      
+      # adjust the pageSize based on the total remaining messages
+      pageSize = min(pageSize, endIndex + 1) 
+
+      if (pageSize == 0):
+        return (@[], PagingInfo(pageSize: pageSize, cursor:pinfo.cursor, direction: pinfo.direction), HistoryResponseError.NONE)
```
Comment (oskarth) [inline waku/v2/protocol/waku_store/waku_store.nim:355]: Can we pull this up into an intermediate variable and return that? Makes debugging easier

File: waku/v2/protocol/waku_store/waku_store.nim [hunk 2]
```diff
@@ -289,54 +289,80 @@ proc paginate*(list: seq[IndexedWakuMessage], pinfo: PagingInfo): (seq[IndexedWa
...
+
+      # set the index of the first message in the page
+      var startIndex = endIndex - pageSize + 1
+
+      # retrieve the messages
+      var retMessages: seq[IndexedWakuMessage]
+      for i in startIndex..endIndex:
+        retMessages.add(msgList[i])
+      return (retMessages, PagingInfo(pageSize : pageSize, cursor : msgList[startIndex].index, direction : pinfo.direction), HistoryResponseError.NONE)
```
Comment (oskarth) [inline waku/v2/protocol/waku_store/waku_store.nim:364]: Ditto

**Review by oskarth — APPROVED** [review]

File: waku/v2/protocol/waku_store/waku_store.nim (hunk 1 above)
Comment (staheri14) [reply waku/v2/protocol/waku_store/waku_store.nim]: Of course, done!

File: waku/v2/protocol/waku_store/waku_store.nim (hunk 2 above)
Comment (staheri14) [reply waku/v2/protocol/waku_store/waku_store.nim]: done!

**Pull request merged**